multihash.go

```diff
@@ -243,21 +243,19 @@ func Decode(buf []byte) (*DecodedMultihash, error) {
 
 // Encode a hash digest along with the specified function code.
 // Note: the length is derived from the length of the digest itself.
-//
-// The error return is legacy; it is always nil.
-func Encode(buf []byte, code uint64) ([]byte, error) {
+func Encode(buf []byte, code uint64) []byte {
 	// FUTURE: this function always causes heap allocs... but when used, this value is almost always going to be appended to another buffer (either as part of CID creation, or etc) -- should this whole function be rethought and alternatives offered?
 	newBuf := make([]byte, varint.UvarintSize(code)+varint.UvarintSize(uint64(len(buf)))+len(buf))
 	n := varint.PutUvarint(newBuf, code)
 	n += varint.PutUvarint(newBuf[n:], uint64(len(buf)))
 
 	copy(newBuf[n:], buf)
-	return newBuf, nil
+	return newBuf
 }
 
 // EncodeName is like Encode() but providing a string name
 // instead of a numeric code. See Names for allowed values.
-func EncodeName(buf []byte, name string) ([]byte, error) {
+func EncodeName(buf []byte, name string) []byte {
```

Comment:
We _should_ be returning an error if the name isn't known. Otherwise, we just default to 0 which is definitely wrong.